bzzeth/bzzeth.go

```diff
@@ -296,7 +311,14 @@ func (b *BzzEth) validateHeader(ctx context.Context, header []byte, req *request
 			return nil, errDuplicateHeader
 		} else {
```

Comment:
Well.. else does two things

1) Tracks weather a particular header is received in a batch
2) sends the header to the delivery channel for processing
         - one is to log in handleNewBlockHeaders for loop
         -  second is to send the header to the light client delivery channel

Reply:
i am just saying  there is no need for `else` since you return in the if clause.

Reply:
@zelig  The return in the if clause happens when a duplicate header is received...
```
else {
			// header is still marked as "yet to be received" and we got that header
			setHeaderAsReceived(req, ch.Address().Hex())
			// This channel is used to track deliveries
			req.c <- header
			return ch, nil
		}
```
The above portion executes when a header is "expected" and " not headerAlreadyReceived" .
Hope this clears .. otherwise i can call you for a quick chat tomorrow...

Reply:
there is a return in the if clause so you do not need else:

```golang
if expected {
		if headerAlreadyReceived {
			// header already received
			return nil, errDuplicateHeader
		} 
                


```

Reply:
ooh . i got it.. fixed it..